pdata/README.md

```diff
@@ -0,0 +1,214 @@
+
+```go
+func (ms NumberDataPoint) IntValue() int64
+func (ms NumberDataPoint) SetIntValue(v int64)
+```
+
+Additionally, the pdata struct MUST provide a type getter for every exposed oneof protobuf field. Name of the getter 
+MUST be `Type` for oneof fields representing the whole protobuf message (1), or `<oneof field>Type` for oneof fields 
+relevant only to a particular field. The function MUST return an `int32` enum type called `<struct name>Type` for (1) 
+and `<struct name><oneof field>Type` for (2). For example:
+
+- `func (ms Metric) Type() MetricType` (1)
+- `func (ms NumberDataPoint) ValueType() NumberDataPointValueType` (2)
+
+The enum constants MUST be called the same as the type with a suffix named after the oneof option, e.g.: 
+
+- `MetricTypeSum` (1)
+- `NumberDataPointValueTypeString` (2)
+
+An unset oneof protobuf value MUST be represented by a pdata constant with `Empty` suffix, e.g.:
+
+- `MetricTypeEmpty` (1)
+- `NumberDataPointValueTypeEmpty` (2)
+
+The pdata enum type SHOULD have `String()` method returning a string value of the corresponding oneof field option. 
+
+### Repeated protobuf fields
+
+Repeated protobuf fields MUST be represented as another pdata struct with `Slice` suffix unless the slice message is 
+represented in pdata through a modified interface, e.g. `pdata.Map`. Each slice MUST have the following methods:
+
+- `Len()`: returns length of the slice.
+- `At(int) <element type>`: returns an element from the slice by its index, can cause "index out of range" panic.
+- `CopyTo(<slice type>)`: copies the slice to another slice of the same type overriding the destination.
+- `EnsureCapacity(int)`: ensures that the slice have at least the requested capacity.
+- `AppendEmpty() <element type>`: adds another slice element with an empty initialized value and returns it.
+- `Sort(func(a, b <element type>) bool) <slice type>`: sorts the slice using the function and returns itself.
+
+### Repeated scalar fields
+
+Repeated scalar protobuf fields MUST be represented as another pdata type wrapping native go slices. Name of the 
+type MUST include name of the primitive data type ending with `Slice`, e.g. `UInt64Slice`. Each pdata type MUST have 
+the following methods:
+
+- `AsRaw() []<primitive type>`: returns copy of the underlying go slice. 
+- `FromRaw(val []<primitive type>)`: copies a go slice into the pdata wrapper instance.
+- `Len()`: returns length of the slice.
+- `At(int) <primitive type>`: returns an element from the slice by its index, can cause "index out of range" panic.
+- `EnsureCapacity(int)`: ensures that the slice have at least the requested capacity.
+- `AppendEmpty() <primitive type>`: adds another slice element with a zero value.
+- `CopyTo(<slice type>)`: copies the slice to another slice of the same type overriding the destination.
+- `MoveTo(<slice type>)`: move the slice to another slice of the same type. The destination is overridden, the 
+  source is set to an empty slice.
```

Comment:
Do we need these detailed docs given that this is all codified in the pdata generator? I am worried that we may forget to keep this up to date when we change any behavior in the generator.
The list of methods also seems incomplete, I can't find `MoveAndAppendTo` or `RemoveIf`.

Reply:
Makes sense. Maybe requiring a particular set of methods is redundant here. Removed them